### ISSUE TITLE
Add webhook to clean up CDN cache of specific pages

### ISSUE THREAD

**Issue by fernandolucchesi** [equinor/energyvision#1286]
https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration
When a document in Sanity is published/updated, you could use a GROQ Webhook to rebuild a specific page by hitting the API for fresh data, while still using the CDN for your 60 second revalidation and static builds.

https://equinor.slack.com/archives/C01A4BGNRMJ/p1663666467602429

**Comment by fernandolucchesi** [comment]
This is ready to be reviewed on https://web-global-development-equinor-web-sites-dev.c2.radix.equinor.com/ and in the sanity.io webhook tab. Changes done in the studio should now be reflected blazingly fast to the web.

@TODO: Add 1782937891 webhooks when this is approved. One for each production dataset.

**Comment by SvSven** [comment]
Changes seem to be taking at least 2 mins for me to be reflected in the web, when viewed on Radix at least. Definitely better than the 20+ mins some people were experiencing, but not sure I would call it blazingly fast 😅

**Comment by fernandolucchesi** [comment]
this solution won't work: https://nextjs.org/docs/going-to-production#caching
<img width="690" alt="image" src="https://user-images.githubusercontent.com/23337266/197414107-6bfb2caa-a57f-4c74-a34f-68727b551f57.png">

**Comment by fernandolucchesi** [comment]
@nilsml @SvSven what do you think we should do here?

**Comment by SvSven** [comment]
Not sure I'm following - does this mean `unstable_revalidate()` does not work because we don't have proper caching set up for this on Radix?

**Comment by fernandolucchesi** [comment]
> Sorry I completely missed this. Has this been enabled on https://studio-global-development-equinor-web-sites-dev.c2.radix.equinor.com?
> 
> Takes a while for changes on the home page to be reflected there; doesn't update immediately on refresh or hard refresh even - hard to say if it's working or not.

yes, it has.. this issue is a hard one to solve, especially because the revalidate should only bypass the revalidate time from staticProps and not give a super hard refresh. However I believe this solves the case when a news is published and it doesn't show up in the latest news component on the homepage, which I believe was the most important thing to get fixed.

**Comment by SvSven** [comment]
Seems to take around 2 mins for newly created articles (with the correct tag) to show up in the news promotion on the front page. Better than 10+ minutes at least 🤷‍♂️

**Comment by fernandolucchesi** [comment]
Do you think it is a good idea to also have the swr solution together with this?

**Comment by SvSven** [comment]
I'm not sure - that did seem to chew into the API usage, and once we look into the new Next 13 features with fetching data on a component level that might be obsolete again.

It's probably fine like this?

**Comment by fernandolucchesi** [comment]
> I'm not sure - that did seem to chew into the API usage, and once we look into the new Next 13 features with fetching data on a component level that might be obsolete again.
> 
> It's probably fine like this?

good point, we should eventually migrate things to server components

**Comment by fernandolucchesi** [comment]
Hi @meols, can you test this on development? The issue is about improving the time that the home page takes to get updated after a new 'news' is published or promoted component is changed etcs

**Comment by meols** [comment]
@fernandolucchesi - the EN home page news promo showed a new general news article within just a few seconds, the NO homepage took app. two minutes to show the new Norwegian news article. Big improvements here :clap:  Also a question; is it expected that the Norwegian should take longer? Not sure if it was part of the testing task, but I also tested the newsroom pages. The English newsroom page showed the new news article quickly (less than a minute). The Norwegian newsroom page still hadn't showed the new news article after **18 minutes**, but when I updated that news article and republished it immediately showed on the newsroom page.

**Comment by fernandolucchesi** [comment]
> @fernandolucchesi - the EN home page news promo showed a new general news article within just a few seconds, the NO homepage took app. two minutes to show the new Norwegian news article. Big improvements here 👏 Also a question; is it expected that the Norwegian should take longer? Not sure if it was part of the testing task, but I also tested the newsroom pages. The English newsroom page showed the new news article quickly (less than a minute). The Norwegian newsroom page still hadn't showed the new news article after **18 minutes**, but when I updated that news article and republished it immediately showed on the newsroom page.

If it worked like it should, everything should be updated within a few seconds.. This is a very tricky one that I am hoping to be sorted out when we upgrade the version of our frontend... I will release it the way it is now..
